packages/test/test-end-to-end-tests/src/test/gc/gcSweepDataStores.spec.ts

```diff
@@ -541,23 +541,245 @@ describeCompat("GC data store sweep tests", "NoCompat", (getTestObjectProvider)
 		});
 
 		it("can run sweep without failing summaries due to local changes", async () => {
-			const { summarizer } =
-				await summarizationWithUnreferencedDataStoreAfterTime(sweepTimeoutMs);
-			await sendOpToUpdateSummaryTimestampToNow(summarizer);
+			const { summarizer } = await summarizationWithUnreferencedDataStoreAfterTime();
 
 			// Summarize. In this summary, the gc op will be sent with the deleted data store id. Validate that
 			// the GC op does not fail summary due to local changes.
 			await assert.doesNotReject(
-				async () => summarizeNow(summarizer),
+				async () => summarize(summarizer),
 				"Summary and GC should succeed in presence of GC op",
 			);
 
 			// Summarize again so that the sweep ready blobs are now deleted from the GC data. Validate that
 			// summarize and GC succeed.
 			await assert.doesNotReject(
-				async () => summarizeNow(summarizer),
+				async () => summarize(summarizer),
 				"Summary and GC should succeed with deleted data store",
 			);
 		});
 	});
+
+	describe("Sweep with summarize failures and retries", () => {
+		const summarizeErrorMessage = "SimulatedTestFailure";
+
+		/**
+		 * This function does the following:
+		 * 1. Overrides the summarize function of the given container runtime to fail until final summarize attempt.
+		 *
+		 * 2. If "pauseInbound" is true, pauses the inbound queue until the final summarize attempt is completed.
+		 *
+		 * 3. Generates and returns a promise which resolves with ISummarizeEventProps on successful summarization.
+		 */
+		async function overrideSummarizeAndGetCompletionPromise(
```

Comment:
If this is identical to the on in the Blobs test file, move to the utils file?

Reply:
I think we should merge these two test files into one. I am going to leave it as-is for now.